src/store/metadata/oxigraph/store.py

```diff
@@ -110,7 +117,73 @@ def get_edition(
+        graph = self.db
+
+        # Use the construct wrappers to pull the raw RDF triples
+        # (as one rdflib.Graph() for each function) and add them
+        # together to create a single Graph of the
+        # data we need.
+        result: Graph = (
+            construct_edition_core(graph, dataset_id, edition_id)
+            + construct_edition_contact_point(graph, dataset_id, edition_id)
+            + construct_edition_themes(graph, dataset_id, edition_id)
+            + construct_edition_keywords(graph, dataset_id, edition_id)
+            + construct_edition_temporal_coverage(graph, dataset_id, edition_id)
+            + construct_edition_table_schema(graph, dataset_id, edition_id)
+            + construct_edition_versions(graph, dataset_id, edition_id)
+        )
+
+        # Serialize the graph into jsonld
+        data = json.loads(result.serialize(format="json-ld"))
+
+        # Use a context file to shape our jsonld, removing long form references
+        data = jsonld.flatten(
+            data, {"@context": constants.CONTEXT, "@type": "dcat:Dataset"}
+        )
+
+        edition_graph = next((x for x in data["@graph"] if "@type" in x.keys()), None)
+        contact_point_graph = next(
+            (x for x in data["@graph"] if "vcard:fn" in x.keys()), None
+        )
+        temporal_coverage_graph = next(
+            (x for x in data["@graph"] if "dcat:endDate" in x.keys()), None
+        )
+        table_schema_graph = next(
+            (x for x in data["@graph"] if "columns" in x.keys()), None
+        )
+        columns_graph = [x for x in data["@graph"] if "datatype" in x.keys()]
+
+        versions_graph = next((x for x in data["@graph"]))
+        # Populate table_schema_graph.columns with column definitions and delete table_schema_graph.column blank node @id
+        for column in table_schema_graph["columns"]:
+            for col in columns_graph:
+                if column["@id"] == col["@id"]:
+                    column["description"] = col["description"]
+                    column["datatype"] = col["datatype"]
+                    column["name"] = col["name"]
+            del column["@id"]
+
+        edition_graph["contact_point"] = {
+            "name": contact_point_graph["vcard:fn"]["@value"],
+            "email": contact_point_graph["vcard:hasEmail"],
+        }
+        edition_graph["temporal_coverage"] = {
+            "start": temporal_coverage_graph["dcat:startDate"]["@value"],
+            "end": temporal_coverage_graph["dcat:endDate"]["@value"],
+        }
+        # Populate editions_graph.table_schema.columns with column definitions and delete editions_graph.table_schema blank node @id
+        edition_graph["table_schema"]["columns"] = table_schema_graph["columns"]
+        del edition_graph["table_schema"]["@id"]
+
+        # Expand version to include issued and modified fields
+        for idx, version in enumerate(edition_graph["versions"]):
```

Comment:
I would consider something like:

- get all graphs in @graph with an id of `/<something>/versions/<something>` into a list and `edition_graph["versions"] = [that list]`

you can probably create the list with a one line lambda tbh, eg something along the lines of:

`versions_graphs = [x for x in data["@graph"] if re.match("some appropriate regex", x["@id"])]`

[some stuff on regular expressions in python](https://www.w3schools.com/python/python_regex.asp) and otherwise just play here until you've got a decent expression: https://regexr.com/

note - my off the cuff syntax might not be 100%, but it'd be along those lines.

Reply:
I absolutely suck at regex, so I've done it like this, which I think should do the job, but let me know if not.